sdl_android_lib/src/com/smartdevicelink/encoder/SdlEncoder.java

```diff
@@ -0,0 +1,159 @@
+package com.smartdevicelink.encoder;
+
+import java.io.IOException;
+import java.io.PipedOutputStream;
+import java.nio.ByteBuffer;
+
+import android.annotation.TargetApi;
+import android.media.MediaCodec;
+import android.media.MediaCodecInfo;
+import android.media.MediaFormat;
+import android.os.Build;
+import android.view.Surface;
+
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+public class SdlEncoder {
+	
+	// parameters for the encoder
+	private static final String _MIME_TYPE = "video/avc"; // H.264/AVC video
+	// private static final String MIME_TYPE = "video/mp4v-es"; //MPEG4 video
+	private static int _FRAME_RATE = 30;
```

Comment:
Remove statics, make constants for defaults and member variables with setters to overwrite the defaults